### ISSUE TITLE
Disappearing Data

### ISSUE THREAD

**Issue by saratheneale** [projectbuendia/client#317]
I backdated the tablet and made a new patient record over 8 days for Patient `Sara90 Sarawr90`
This all takes place on "June 18." Around 16:48 PST
At 7:49 or 7:48 I hit save on New Round (with data that's mostly 59 input).
At 7:49 I noticed the data didn't appear.
At 7:49 the sync happened and the data came back. 

Attached is an adb log from 7:45 to 7:50.

**Comment by saratheneale** [comment]
EDIT: last file was way more verbose. Here's actually 745 - 750: 

[745-750.txt](https://github.com/projectbuendia/client/files/3308306/745-750.txt)

**Comment by saratheneale** [comment]
Other notes via @sdspikes 
https://docs.google.com/document/d/1b3ZSNUgy9uuCz-EZLB8AwFCHGPm7bVsd96-gfNPkJT8/edit

```
Install app fresh or wipe data

Go to user
Create new patient

Create new chart item for today, fully filled out
Create new chart item for a few days back, fully filled out.  It flashes on the table then disappears.  If you wait about 1 min or leave and come back it eventually re-appears

Some examples of patients and chart dates that did the disappearing act:
Jane Goodall, Jun 9
Ada Lovelace Jun 11 -- ones added later on (earlier dates) it worked fine  

(Note: I did the same process a few times and it did appear right away and never disappeared, so it’s not 100%.  Seems to happen maybe 25-50% of the time when following these directions.  Some chance that it matters whether/how you fill out specific fields?  It never happens when adding data to existing patients AFAICT)



Slightly different but possibly related bug:
Temperence Brennan - June 10 -- the new row appeared right away but only had condition and notes.  After a minute or so, the rest filled in (no user interaction, was writing down result)

```

**Comment by zestyping** [comment]
@saratheneale Do you know how far the tablet's clock was backdated by?

**Comment by zestyping** [comment]
@saratheneale Do you know what time zone the tablet was in, so I know how to interpret 07:48?  Also do you know what the real time was when you saved this new round (07:48 on the tablet was what time in real life?)

**Comment by zestyping** [comment]
On the server side, I see that at 2019-06-19T23:04:28Z, patient `Sarawr90` was created, with patient UUID `d4db3595-d98f-44f4-9e12-5562363f8d8d`.

These are the request logs on the server from approximately 2019-06-19T22:25Z to 2019-06-20T01:00Z:
[issue-317-span.txt](https://github.com/projectbuendia/client/files/3312401/issue-317-span.txt)

And this is a `grep -C 2` of this span of time searching for the patient UUID:
[issue-317-patient.txt](https://github.com/projectbuendia/client/files/3312402/issue-317-patient.txt)

**Comment by saratheneale** [comment]
It was IRL around 5pm. So probably at 16:48 PST. 

> @saratheneale Do you know what time zone the tablet was in, so I know how to interpret 07:48? Also do you know what the real time was when you saved this new round (07:48 on the tablet was what time in real life?)

**Comment by saratheneale** [comment]
I ran into this again after filling out an Ebola Test on a patient. I filled out the form with 13 / 13 and saved it. 
The form saved and went back to the Patient Record. However, the Ebola Test field was still blank. 
So 30 seconds later, I went in and did it again with similar data. On this second save, the chart was still blank and then immediately updated with the saved data. 

I tried running this with on the minute changes right before and after. But I can't reproduce. 
Another test case to try might be right before and after a server sync? I don't have the tools to know when the next server sync is though. Just an idea.
